plugins/services/src/js/service-configuration/ServiceStorageConfigSection.js

```diff
@@ -31,160 +24,114 @@ class ServiceStorageConfigSection extends ServiceConfigBaseSectionDisplay {
-                volume.type.push("External");
-              } else {
-                volume.isHostVolume = true;
-                volume.type.push("Host", "Volume");
-              }
-
-              if (appVolume.external != null) {
-                volume.name = appVolume.external.name;
-              }
-
-              if (appVolume.hostPath != null) {
-                // Set this flag to true so that we render the hostPath column.
-                shouldDisplayHostPath = true;
-              }
-
-              volume.hostPath = getDisplayValue(
-                appVolume.hostPath,
-                !volume.isHostVolume
-              );
-
-              return volume;
-            });
-
-            if (shouldDisplayHostPath) {
-              columns.push({
-                heading: getColumnHeadingFn("Host Path"),
-                prop: "hostPath",
-                className: getColumnClassNameFn(),
-                sortable: true
-              });
-            }
-
-            if (onEditClick) {
-              columns.push({
-                heading() {
-                  return null;
-                },
-                className: "configuration-map-action",
-                prop: "edit",
-                render() {
-                  return (
-                    <a
-                      className="button button-link flush table-display-on-row-hover"
-                      onClick={onEditClick.bind(null, "volumes")}
-                    >
-                      Edit
-                    </a>
-                  );
-                }
-              });
-            }
-
-            return (
-              <Table
-                key="service-volumes"
-                className="table table-flush table-borderless-outer table-borderless-inner-columns vertical-align-top table-break-word table-fixed-layout flush-bottom"
-                columns={columns}
-                data={volumesData}
-              />
-            );
-          }
         }
       ]
     };
+
+    volumes.forEach((volume, index) => {
+      const configHeading = {
+        heading: null,
+        headingLevel: 2
+      };
+      const configName = {
+        key: null,
+        label: "NAME"
+      };
+      const configContainerPath = {
+        key: null,
+        label: "CONTAINER PATH"
+      };
+      const configSize = {
+        key: null,
+        label: "SIZE",
+        transformValue: null
+      };
+
+      if (volume.persistent != null && volume.persistent.profileName != null) {
+        // DSS
+        configHeading.heading = "DC/OS Storage Service";
+        configName.key = `container.volumes.${index}.persistent.profileName`;
+        configName.name = "PROFILE NAME";
+        configContainerPath.key = `container.volumes.${index}.containerPath`;
+        configSize.key = `container.volumes.${index}.persistent.size`;
+        configSize.transformValue = this.getVolumeSizeValue;
+
+        config.values.push(
```

Comment:
I think instead of mutating `config.values` you should do a small refactor to concat this result at object creation.

```

Reply:
Ok sounds good. thanks.

Reply:
I did a bit different to keep a bit cleaner. but thanks for the suggestion.